migration-cstor-mayastor/distributeddb-restore.md

```diff
@@ -0,0 +1,345 @@
+
+### Access a Cassandra pod using cqlsh and check the data
+
+- You can use the following command to access the Cassandra pods. This command establishes a connection to the Cassandra database running on pod `cassandra-1`:
+
+```
+cqlsh -u <enter-your-user-name> -p <enter-your-password> cassandra-1.cassandra-headless.cassandra.svc.cluster.local 9042
+```
+- The query results should display the data you backed up from cStor. In your output, you're expecting to see the data you backed up.
+
+```
+cassandra@cqlsh> USE openebs;
+```
+
+```
+cassandra@cqlsh:openebs> select * from openebs.data;
+```
+
+```
+ replication | appname   | volume
+-------------+-----------+--------
+           3 | cassandra |  cStor
+
+(1 rows)
+```
+
+- After verifying the data, you can exit the Cassandra shell by typing `exit`.
+
+
+
+
+
+
+
+### Modify your Cassandra StatefulSet YAML to use the mayastor-single-replica storage class
+
+- Before making changes to the Cassandra StatefulSet YAML, create a backup to preserve the existing configuration by running the following command:
+
+```
+kubectl get sts cassandra -n cassandra -o yaml > cassandra_sts_backup.yaml
+```
+
+- You can modify the Cassandra StatefulSet YAML to change the storage class to `mayastor-single-replica`. Here's the updated YAML:
+```
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  annotations:
+    meta.helm.sh/release-name: cassandra
+    meta.helm.sh/release-namespace: cassandra
+  labels:
+    app.kubernetes.io/instance: cassandra
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/name: cassandra
+    helm.sh/chart: cassandra-10.5.3
+    velero.io/backup-name: cassandra-backup-19-09-23
+    velero.io/restore-name: cassandra-restore-19-09-23
+  name: cassandra
+  namespace: cassandra
+spec:
+  podManagementPolicy: OrderedReady
+  replicas: 3
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: cassandra
+      app.kubernetes.io/name: cassandra
+  serviceName: cassandra-headless
+  template:
+    # ... (rest of the configuration remains unchanged)
+  updateStrategy:
+    type: RollingUpdate
+  volumeClaimTemplates:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      creationTimestamp: null
+      labels:
+        app.kubernetes.io/instance: cassandra
+        app.kubernetes.io/name: cassandra
+      name: data
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 3Gi
+      storageClassName: mayastor-single-replica # Change the storage class here
+      volumeMode: Filesystem
+```
+
+- Apply the modified YAML to make the changes take effect:
+
+```
+kubectl apply -f cassandra_sts_modified.yaml
+```
+
+
+
+### Delete the Cassandra StatefulSet with the --cascade=false flag
+
+Delete the Cassandra StatefulSet while keeping the pods running without controller management:
+
+```
+kubectl delete sts cassandra -n cassandra --cascade=orphan
```

Comment:
we have to use either false or orphan, change this flag to false or change line 312 to --cascade=orphan.